File: Java/Desafio3 - GFT/src/model/Caminhao.java
package model;

public class Caminhao extends Veiculo {
	
	private int altura;
	private String uso;
	
	public int getAltura() {
		return altura;
	}
	public void setAltura(int altura) {
		this.altura = altura;
	}
	public String getUso() {
		return uso;
	}
	public void setUso(String uso) {
		this.uso = uso;
	}
	
	public void abastecer(float qtd) {
		int litros = (int) qtd;
		litros = getLitrosCombustivel() + litros;
		if (litros > 275) {
			System.out.println("Seu tanque só comporta 275 litros");
			setLitrosCombustivel(275);
		} else {
				setLitrosCombustivel(litros);
		}
		System.out.println("Litros de combustível após abastecer: " + getLitrosCombustivel() + " litros");
	}
	
	@Override
	public void acelerar() {
		//verificar se está ligado
		if (getLigado() == true) {
			setVelocidade(getVelocidade() + 15);
			setLitrosCombustivel(getLitrosCombustivel() - 5);
			System.out.println("Velocidade após aceleração: " + getVelocidade() + " km/h" + "\nLitros de combustível após aceleração: " + getLitrosCombustivel() + " litros");
		} else {
			System.out.println("Não é possível acelerar com o caminhão desligado");
		}
	}

}

File: C-Sharp/Desafio03/Desafio03/com.gft.model/Caminhao.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Desafio03.com.gft.model
{
    class Caminhao:Veiculo
    {
        private int _altura;
        private String _uso;

        public int Altura { get => _altura; set => _altura = value; }
        public string Uso { get => _uso; set => _uso = value; }

		public void abastecer(float qtd)
		{
			int litros = (int)qtd;
			litros = LitrosCombustivel + litros;
			if (litros > 275)
			{
				Console.WriteLine("Seu tanque só comporta 275 litros");
				LitrosCombustivel = 275;
			}
			else
			{
				LitrosCombustivel = litros;
			}
			Console.WriteLine("Litros de combustível após abastecer: " + LitrosCombustivel + " litros");
		}

		public override void acelerar()
		{
			//verificar se está ligado
			if (IsLigado == true)
			{
				Velocidade = Velocidade + 15;
				LitrosCombustivel = LitrosCombustivel - 5;
				Console.WriteLine("Velocidade após aceleração: " + Velocidade + " km/h" + "\nLitros de combustível após aceleração: " + LitrosCombustivel + " litros");
			}
			else
			{
				Console.WriteLine("Não é possível acelerar com o caminhão desligado");
			}
		}
	}
}
